lib/utils.js

```diff
@@ -52,8 +56,68 @@ function resolveRepositoryPath(options, owner, repo) {
   return repPath;
 }
 
+module.exports._caseInsensitiveFS = undefined;
+
+/**
+ * Returns true if the file system where the current executable was
+ * started from is case-insensitive, otherwise returns false.
+ */
+async function isCaseInsensitiveFS() {
+  if (typeof module.exports._caseInsensitiveFS === 'undefined') {
+    let lcStat;
+    let ucStat;
+    try {
+      lcStat = await fse.stat(process.execPath.toLowerCase());
+    } catch (err) {
+      lcStat = false;
+    }
+    try {
+      ucStat = await fse.stat(process.execPath.toUpperCase());
+    } catch (err) {
+      ucStat = false;
+    }
+    if (lcStat && ucStat) {
+      module.exports._caseInsensitiveFS = lcStat.dev === ucStat.dev && lcStat.ino === ucStat.ino;
+    } else {
+      module.exports._caseInsensitiveFS = false;
+    }
+  }
+  return module.exports._caseInsensitiveFS;
+}
+
+/**
+ * Test whether or not a file system entry exists at `pathToTest` with the same case as specified.
+ *
+ * @param {string} parentDir
+ * @param {string} pathToTest
+ */
+async function pathExists(parentDir, pathToTest) {
+  if (!await isCaseInsensitiveFS()) {
+    return fse.pathExists(path.join(parentDir, pathToTest));
+  }
+
+  let parent = parentDir;
+
+  const names = pathToTest.split('/').filter(el => el !== '');
```

Comment:
does the `split('/')` work on windows?

Reply:
IMO better to use [`path.sep`](https://nodejs.org/api/path.html#path_path_sep)

Reply:
@tripodsan @filmaj Initially I had `path.sep` but changed to `/`. `pathToTest` is taken from the url, hence it contains `/`.